test/src/main/java/org/apache/accumulo/test/ShellServerIT.java

```diff
@@ -182,14 +186,19 @@ public String getErrorMessage() {
     public TestOutputStream output;
     public StringInputStream input;
     public Shell shell;
+    public LineReader reader;
+    public Terminal terminal;
 
     TestShell(String user, String rootPass, String instanceName, String zookeepers, File configFile)
         throws IOException {
       ClientInfo info = ClientInfo.from(configFile.toPath());
       // start the shell
       output = new TestOutputStream();
       input = new StringInputStream();
-      shell = new Shell(new ConsoleReader(input, output));
+      terminal = new DumbTerminal(input, output);
+      terminal.setSize(new Size(80, 24));
+      reader = LineReaderBuilder.builder().terminal(terminal).build();
+      shell = new Shell(reader);
```

Comment:
I wonder if it would make more sense to pass in "terminal" instead of just the reader?

Reply:
I am unsure if it would change anything since the terminal gets attached to the reader. We could just pass in "terminal" and then build the reader inside `Shell.java` so keep things consistent there. Not sure what that would gain ultimately.

Reply:
The gain is just more intuitive API semantics (possibly) and maybe exposing a more stable jline API... a terminal being a high level object and line reader a lower level component.

Reply:
I will look into it further once I get through the rest of the comments.

Reply:
I think your other comment explained that LineReader was more the primary object to interact with, not the terminal. If that's the case, then my suggestion here probably is fine to ignore.